### PR TITLE
Make sure to add user with "adduser"

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
I don't get what `docker_rules` is doing behind the scenes when building. But I can see it's possible to make the home directory whose owner is the user (rather than root).

**Which issue(s) this PR fixes**:

Part of https://github.com/pipe-cd/pipe/issues/987

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -133,7 +133,7 @@ container_pull(
     digest = "sha256:e357a240b1e8668c851a1dcb3e3244b4e037aab56029f394dda6925070d473a9",
     registry = "gcr.io",
     repository = "pipecd/piped-base",
-    tag = "0.1.0",
+    tag = "0.1.1",
```
Comment (nghialv) [inline WORKSPACE:136]: This should be updated in another PR after this image was pushed.
And the digest must be updated too.

Comment (nakabonne) [reply WORKSPACE:136]: Ah, I got wrong! Thanks.

File: dockers/piped-base/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 FROM alpine:3.10.1
 
-ENV PIPED_BIN_DIR=/usr/local/piped
+ENV PIPED_BIN_DIR=/home/pipecd
```
Comment (nghialv) [inline dockers/piped-base/Dockerfile:3]: nit: `/home/pipecd/tools`

File: WORKSPACE (hunk 1 above)
Comment (nakabonne) [reply WORKSPACE:136]: Is it no problem to keep `dockers/piped-base/DOCKER_BUILD` as is?

File: dockers/piped-base/Dockerfile (hunk 2 above)
Comment (nghialv) [reply dockers/piped-base/Dockerfile:3]: Because we will have `.ssh` directory in this too.

Comment (nakabonne) [reply dockers/piped-base/Dockerfile:3]: Right, it should be aggregated into one place.

File: WORKSPACE (hunk 1 above)
Comment (nghialv) [reply WORKSPACE:136]: Yes, that file is for defining the version of that docker image. And so that is ok.

Comment (nakabonne) [reply WORKSPACE:136]: 👍

File: dockers/piped-base/Dockerfile [hunk 3]
```diff
@@ -27,4 +27,8 @@ RUN \
     /installer/install-terraform.sh && \
     # Delete installer directory.
     rm -rf /installer && \
-    rm -f /var/cache/apk/*
+    rm -f /var/cache/apk/* && \
+    addgroup -S -g 1000 pipecd && \
+    adduser -S -u 1000 -G pipecd -h /home/pipecd pipecd
+
+USER pipecd
```
Comment (nghialv) [inline dockers/piped-base/Dockerfile:34]: I think we should move lines 31-34 to before installing the tools (before line 11).

Comment (nakabonne) [reply dockers/piped-base/Dockerfile:34]: I see, gonna give it a try.

Comment (nakabonne) [reply dockers/piped-base/Dockerfile:34]: Seems to be unable to perform apk with pipecd user. It may be good to keep the order as is and just add `-p` option to mkdir command.
```
ERROR: Unable to lock database: Permission denied
ERROR: Failed to open apk database: Permission denied
```

Comment (nakabonne) [reply dockers/piped-base/Dockerfile:34]: I could see the owner of the home directory is pipecd even if created in the build phase with root permission.
```
/ $ ls -la /home/pipecd/
total 20
drwxr-sr-x    1 pipecd   pipecd        4096 Oct 23 01:47 .
drwxr-xr-x    1 root     root          4096 Oct 23 01:47 ..
-rw-------    1 pipecd   pipecd          44 Oct 23 01:48 .ash_history
drwxr-xr-x    2 root     root          4096 Oct 23 01:47 tools
```

**Comment by nakabonne** [comment]
Hmm... I don't get what's going on when docker testing yet. Seems that we needed to dig deeper into Kapetanios's docker plugin.

```
/bin/sh: can't open '/tmp/agent851078611/runner': Permission denied
```

**Comment by nakabonne** [comment]
I could see the owner of `/home/tools` is changed to pipecd user as well:
```bash
❯ docker run --rm -it piped-base-test:0.5 sh
/ $ ls -la /home/pipecd/
total 20
drwxr-sr-x    1 pipecd   pipecd        4096 Oct 23 02:21 .
drwxr-xr-x    1 root     root          4096 Oct 23 02:21 ..
-rw-------    1 pipecd   pipecd          41 Oct 23 02:22 .ash_history
drwxr-sr-x    2 pipecd   pipecd        4096 Oct 23 02:21 tools
```

**Comment by nakabonne** [comment]
@nghialv Could you do the last check?

**Comment by nghialv** [comment]
Ok. Let's merge this pull request.
